pkg/core/redis/pool.go

```diff
@@ -1,20 +1,33 @@
 package redis
 
 import (
+	"errors"
 	"fmt"
 	"time"
 
+	"github.com/FZambia/sentinel"
 	"github.com/gomodule/redigo/redis"
 )
 
+// When sentinel is enabled, host and port will be ignored
 type Configuration struct {
-	Host     string `envconfig:"HOST"`
-	Port     int    `envconfig:"PORT" default:"6379"`
-	Password string `envconfig:"PASSWORD"`
-	DB       int    `envconfig:"DB" default:"0"`
+	Host     string         `envconfig:"HOST"`
+	Port     int            `envconfig:"PORT" default:"6379"`
+	Password string         `envconfig:"PASSWORD"`
+	DB       int            `envconfig:"DB" default:"0"`
+	Sentinel SentinelConfig `envconfig:"SENTINEL"`
+}
+
+type SentinelConfig struct {
+	Enabled    bool     `envconfig:"ENABLED"`
+	Addrs      []string `envconfig:"ADDRS"`
+	MasterName string   `envconfig:"MASTER_NAME" default:"mymaster"`
```

Comment:
I don't think we should provide a default value here. It seems this name need to be same as configuration of Redis, so the default is unlikely to be useful.